src/dune_pkg/tar.ml

```diff
@@ -46,3 +46,14 @@ let extract ~archive ~target =
     Path.rename target_in_temp target;
     Ok ()
 ;;
+
+let load_or_untar ~target ~archive =
+  match Path.exists target with
```

Comment:
What if the tar file is updated remotely?

Reply:
I still don't understand the point of this check

Reply:
How do you know it's the same file though? If you're relying on the digest, you should at least check that.

Reply:
Here, the name of the target (a directory) is the same as the digest of the archive (compute above). I have moved the code directly where it is used as this function would be a source of errors otherwise.